src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -2205,5 +2216,66 @@ private static List<CompletionResult> CompleteFileNameAsCommand(CompletionContex
 
             return result;
         }
+
+        private static List<CompletionResult> CompleteUsingKeywords(int cursorOffset, Token[] tokens, ref int replacementIndex, ref int replacementLength)
+        {
+            var result = new List<CompletionResult>();
+            Token tokenBeforeCursor = null;
+            Token tokenAtCursor = null;
+
+            for (int i = tokens.Length - 1; i >= 0; i--)
+            {
+                if (tokens[i].Extent.EndOffset < cursorOffset && tokens[i].Kind != TokenKind.LineContinuation)
+                {
+                    tokenBeforeCursor = tokens[i];
+                    break;
+                }
+                else if (tokens[i].Extent.StartOffset <= cursorOffset && tokens[i].Extent.EndOffset >= cursorOffset && tokens[i].Kind != TokenKind.LineContinuation)
+                {
+                    tokenAtCursor = tokens[i];
+                }
+            }
+
+            if (tokenBeforeCursor is not null && tokenBeforeCursor.Kind == TokenKind.Using)
+            {
+                string wordToComplete = null;
+                if (tokenAtCursor is not null)
+                {
+                    replacementIndex = tokenAtCursor.Extent.StartOffset;
+                    replacementLength = tokenAtCursor.Extent.Text.Length;
+                    wordToComplete = tokenAtCursor.Text;
+                }
+                else
+                {
+                    replacementIndex = cursorOffset;
+                    replacementLength = 0;
+                }
+
+                foreach (var keyword in s_usingKeywords)
+                {
+                    if (string.IsNullOrEmpty(wordToComplete) || keyword.StartsWith(wordToComplete, StringComparison.OrdinalIgnoreCase))
+                    {
+                        string toolTip = ResourceManagerCache.GetResourceString(typeof(CompletionAnalysis).Assembly,
+                            "System.Management.Automation.resources.TabCompletionStrings",
+                            $"{keyword}KeywordDescription");
```

Comment:
I see you follow the file style but I'd prefer we follow our common rule and use strong typed resources. 
We could create a private helper with case selection.
```c#
private string GetUsingTips(string keyword) => keyword switch
{
    assembly => TabCompletionStrings.AssemblyKeywordDescription,